packages/runtime/container-runtime/src/summary/summarizer.ts

```diff
@@ -132,10 +116,12 @@ export class Summarizer extends EventEmitter implements ISummarizer {
 		};
 
 		const resolvedContainer = await loader.resolve(request);
-		const fluidObject = await requestFluidObject<FluidObject<ISummarizer>>(resolvedContainer, {
-			url: "_summarizer",
-		});
-		if (fluidObject.ISummarizer === undefined) {
+		const fluidObject: FluidObject<ISummarizer> | undefined = resolvedContainer.getEntryPoint
+			? await resolvedContainer.getEntryPoint?.()
+			: await requestFluidObject<FluidObject<ISummarizer>>(resolvedContainer, {
+					url: "_summarizer",
+			  });
```

Comment:
@agarwal-navin would like your thoughts on this piece (and the analogous one in `TestSummaryUtils.ts`. Since this PR removes the capacity of the summarizer to respond to a request to the `_summarizer` url, is the fallback logic here necessary? Can this code be running next to code where a summarizer exists that *would* respond to that request?

Reply:
Good question. I don't think the back compat code is necessary because the summarizer and container runtime would be running the same version.
Maybe test the following combinations:
1. Container runtime N (with this change) and loader N-1 in the client. I believe we already have tests that will cover this.
2. Container runtime N in client1 running with container runtime N-1 in client 2. We have examples of these kind of tests. See `describeInstallVersions`.

Reply:
glad this stayed. we need it as the loader could be older than the containerRuntime, so the loader may not have getEntryPoint yet, so there is no path to use it even if the containerRuntime supports it, which will always be the case if this code is running.

Reply:
once the loader layer saturates with getEntryPoint, only then can we remove the fallback.

Reply:
(Mostly for my education). But this code is part of the container runtime layer code, right? Whatever version the loader is, if it loaded a version of the container runtime that has this change, doesn't `loader.resolve()` in line 118 return a (summarizer) container that is guaranteed to have `getEntryPoint`?